frontend/src/screens/wallet/Send.tsx

```diff
@@ -16,98 +10,123 @@ import {
 import { useChannels } from "src/hooks/useChannels";
 
 import { useInfo } from "src/hooks/useInfo";
-import { copyToClipboard } from "src/lib/clipboard";
-import { PayInvoiceResponse } from "src/types";
-import { request } from "src/utils/request";
+
+// email regex: https://emailregex.com/
+// modified to allow _ in subdomains
+const LIGHTNING_ADDRESS_REGEX =
+  /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-_0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+const BOLT11_REGEX = /^(lnbc|lntbs)([0-9a-z]+)$/i;
 
 export default function Send() {
   const { hasChannelManagement } = useInfo();
   const { data: balances } = useBalances();
   const { data: channels } = useChannels();
-
   const { toast } = useToast();
-  const [isLoading, setLoading] = React.useState(false);
-  const [invoice, setInvoice] = React.useState("");
-  const [invoiceDetails, setInvoiceDetails] = React.useState<Invoice | null>(
-    null
-  );
-  const [payResponse, setPayResponse] =
-    React.useState<PayInvoiceResponse | null>(null);
-  const [paymentDone, setPaymentDone] = React.useState(false);
 
-  if (!balances || !channels) {
-    return <Loading />;
-  }
+  const [amount, setAmount] = React.useState<number>(0);
+  const [recipient, setRecipient] = React.useState<string>("");
+  const [lnurlDetails, setLnurlDetails] = React.useState<
+    Invoice | LightningAddress
+  >();
 
-  const handleContinue = () => {
-    try {
-      setInvoiceDetails(new Invoice({ pr: invoice }));
-    } catch (error) {
-      toast({
-        variant: "destructive",
-        title: "Invalid Payment Request",
-      });
-      console.error(error);
-      setInvoice("");
-    }
+  const paste = async () => {
+    const text = await navigator.clipboard.readText();
+    setRecipient(text.trim());
   };
 
-  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     try {
-      setLoading(true);
-      const payInvoiceResponse = await request<PayInvoiceResponse>(
-        `/api/payments/${invoice}`,
-        {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
+      if (LIGHTNING_ADDRESS_REGEX.test(recipient)) {
+        const lnAddress = new LightningAddress(recipient);
+        await lnAddress.fetch();
+        if (!lnAddress.lnurlpData) {
+          throw new Error("invalid lightning address");
         }
-      );
-      if (payInvoiceResponse) {
-        setPayResponse(payInvoiceResponse);
-        setPaymentDone(true);
-        setInvoice("");
-        toast({
-          title: "Successfully paid invoice",
-        });
+        setLnurlDetails(lnAddress);
+      } else if (BOLT11_REGEX.test(recipient)) {
+        const invoice = new Invoice({ pr: recipient });
+        setAmount(invoice.satoshi);
+        setLnurlDetails(invoice);
```

Comment:
hmm, the invoice is not an LNURL, it's a bolt11 invoice. At the least we should rename this.

Reply:
Umm, it can be both an Invoice or Lightning Address so I used the name... I'll check your other comments and refactor this a bit and will see what we can do here :+1:

Reply:
But an invoice is not an LNURL, why did you choose this name?